src/lib/vocab/role/roleMutationHelpers.ts

```diff
@@ -0,0 +1,22 @@
+import {writable} from '@feltcoop/svelte-gettable-stores';
+import {Logger} from '@feltcoop/felt/util/log.js';
+
+import type {WritableUi} from '$lib/ui/ui';
+import type {Role} from '$lib/vocab/role/role';
+
+const log = new Logger('[roleMutationHelpers]');
+
+export const stashRole = (ui: WritableUi, $role: Role): void => {
+	log.trace('stashing role', $role.role_id);
+	const {rolesById, roles} = ui;
+	const {role_id} = $role;
+	let role = rolesById.get(role_id);
+	if (role) {
+		role.set($role);
+	} else {
+		role = writable($role);
+		rolesById.set(role_id, role);
+		roles.get().value.push(role);
+		roles.mutate();
```

Comment:
```suggestion
		roles.mutate(($v) => $v.push(role));
```

Been using the `_.mutate();` pattern only for batching or conditional mutation.